### ISSUE TITLE
No failure if unexecuted script is less than executed scripts

### ISSUE THREAD

**Issue by magitech** [DbUp/DbUp#198]
Scripts will run in lexicographic order but do not respect if a later script has already been run. For example, if the database has been upgraded with Script1.sql then Script0.sql is added DbUp will attempt to run Script0.sql on the next upgrade. My expectation would be such script would not be executed and warnings/errors would be logged. As a nice to have create an override to disable skipping to allow for backwards compatibility.

**Comment by JakeGinnivan** [comment]
Completely agree, this would be a great addition.

Would you be able to submit a PR for v4? (release/4.0.0 branch)

**Comment by robvanuden** [comment]
I can pick this up if you're ok with that!

**Comment by droyad** [comment]
I disagree with this, say two people are working on a branch, the first commits `Script0001 - Foo` the other `Script0001 - Bar`. If the former is merged to `master` before the latter, the latter script will never run. The first one could have gone to production. The second person would need to check to see if that is the case and then rename the script.

This scenario happens all the time for us and the current situation all works pretty seamlessly as the scripts usually modify different parts of the system.

If this is something people want, it should be opt-in

**Comment by dazinator** [comment]
I agree with @droyad, it also means the schemaversion table would end up looking different on different environments, for example some would have executed script 0 and others would not have. This kind of violates the principles behind DbUp in my opinion, and therefore having it an "opt in" feature would be better.

**Comment by wozzo** [comment]
It is opt in. The additional config val needs setting or it does nothing.

**Comment by droyad** [comment]
@wozzo Awesome

**Comment by mjauernig** [comment]
Seems to be done already. If still needed, please reopen.
